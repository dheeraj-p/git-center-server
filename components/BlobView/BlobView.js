import { Fragment } from 'react';
import { Prism as SyntaxHighlighter } from 'react-syntax-highlighter';
import { prism } from 'react-syntax-highlighter/dist/cjs/languages/prism';

export default function BlobView({ path, content }) {
  const contentBuffer = Buffer.from(content.data);
  return (
    <Fragment>
      <p className='text-muted'>{path}</p>
      <div>
        <SyntaxHighlighter showLineNumbers={true} style={prism}>
          {contentBuffer.toString()}
        </SyntaxHighlighter>
      </div>
    </Fragment>
  );
}
