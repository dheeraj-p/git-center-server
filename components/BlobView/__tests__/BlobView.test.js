import { shallow } from 'enzyme';
import BlobView from '../BlobView';

describe('BlobView', () => {
  it('should render properly', () => {
    const content = {
      data: 'test content data',
    };
    const wrapper = shallow(<BlobView path='/test/path' content={content} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
