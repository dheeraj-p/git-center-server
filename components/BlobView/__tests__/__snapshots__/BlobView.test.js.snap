// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlobView should render properly 1`] = `
"<Fragment>
  <p className=\\"text-muted\\">
    /test/path
  </p>
  <div>
    <SyntaxHighlighter showLineNumbers={true} style={[undefined]}>
      test content data
    </SyntaxHighlighter>
  </div>
</Fragment>"
`;
